crypto/src/signature/ed25519.rs

```diff
@@ -126,7 +117,7 @@ mod test {
                 signature.as_slice().as_ptr(),
                 MESSAGE_1.as_ptr(),
                 MESSAGE_1.len() as u64,
-                p.payload().as_ptr(),
+                p.to_bytes().as_ptr(),
```

Comment:
Will the lifetimes be all right here?
It seems that this will pass a pointer to a temporary value that will get destroyed before the function is called (I don't think there's lifetime extension magic like in C++ for cases like these?)

Reply:
Hm, interesting. Fixed.